client/my-sites/domains/domain-search/index.tsx

```diff
@@ -53,62 +53,94 @@ import {
 	getSelectedSiteSlug,
 } from 'calypso/state/ui/selectors';
+import type { Context } from '@automattic/calypso-router';
+import type { DomainSuggestion, SiteDetails } from '@automattic/data-stores';
 
 import './style.scss';
 import 'calypso/my-sites/domains/style.scss';
 
-class DomainSearch extends Component {
-	static propTypes = {
-		basePath: PropTypes.string.isRequired,
-		context: PropTypes.object.isRequired,
-		domainsWithPlansOnly: PropTypes.bool.isRequired,
-		isSiteUpgradeable: PropTypes.bool,
-		productsList: PropTypes.object.isRequired,
-		selectedSite: PropTypes.object,
-		selectedSiteId: PropTypes.number,
-		selectedSiteSlug: PropTypes.string,
-		domainAndPlanUpsellFlow: PropTypes.bool,
-		isDomainAndPlanPackageFlow: PropTypes.bool,
-	};
-
+type DomainSearchProps = {
+	basePath: string;
+	context: Context;
+	domainsWithPlansOnly: boolean;
+	isSiteUpgradeable: boolean | null;
+	productsList: object;
+	selectedSite?: SiteDetails | null;
+	selectedSiteId?: number;
+	selectedSiteSlug: string | null;
+	domainAndPlanUpsellFlow?: boolean;
+	isDomainAndPlanPackageFlow?: boolean;
+	cart: ObjectWithProducts;
+	shoppingCartManager: UseShoppingCart;
+	isAddNewDomainContext: boolean;
+	setCurrentFlowName: ( flowName: string ) => void;
+	recordAddDomainButtonClick: (
+		domainName: string,
+		section: string,
+		position: number,
+		isPremium?: boolean
+	) => void;
+	recordRemoveDomainButtonClick: ( domainName: string ) => void;
+	isSiteOnFreePlan?: boolean;
+	isSiteOnMonthlyPlan: boolean;
+	isDomainUpsell: boolean;
+	currentRoute: string;
+	isFromMyHome: boolean;
+	translate: ReturnType< typeof useTranslate >;
+	isManagingAllDomains: boolean;
+	isEcommerceSite: boolean;
+	preferredView: ReturnType< typeof getSiteOption >;
+	wpAdminUrl: ReturnType< typeof getSiteAdminUrl >;
+};
+
+class DomainSearch extends Component< DomainSearchProps > {
 	isMounted = false;
 
 	state = {
 		domainRegistrationAvailable: true,
 		domainRegistrationMaintenanceEndTime: null,
 	};
 
-	handleDomainsAvailabilityChange = ( isAvailable, maintenanceEndTime = null ) => {
+	handleDomainsAvailabilityChange = (
+		isAvailable: boolean,
+		maintenanceEndTime: number | null = null
+	) => {
 		this.setState( {
 			domainRegistrationAvailable: isAvailable,
 			domainRegistrationMaintenanceEndTime: maintenanceEndTime,
 		} );
 	};
 
-	handleAddRemoveDomain = ( suggestion, position ) => {
+	handleAddRemoveDomain = ( suggestion: DomainSuggestion, position: number ) => {
 		if ( ! hasDomainInCart( this.props.cart, suggestion.domain_name ) ) {
 			this.addDomain( suggestion, position );
 		} else {
 			this.removeDomain( suggestion );
 		}
 	};
 
-	handleAddMapping = ( domain ) => {
-		const domainMappingUrl = domainUseMyDomain( this.props.selectedSiteSlug, {
-			domain,
-			initialMode: useMyDomainInputMode.transferOrConnect,
-		} );
-		this.isMounted && page( domainMappingUrl );
+	handleAddMapping = ( domain: string ) => {
+		if ( this.props.selectedSiteSlug ) {
```

Comment:
This check ensures `selectedSiteSlug` is not `null` before proceeding. I'm not thrilled with this approach, and would love some feedback.  I'd like to be able to more confidently assert that it will never be `null` (rather than just appeasing TS as I'm doing now), or add something to ensure a `null` value is handled gracefully.

Reply:
I see :( 

Let's go the next way:

```
// Just TS typing fix, we always have selectedSiteSlug
if ( !this.props.selectedSiteSlug ) {
	return;
}

.... rest of teh function
```

Cooment will be very useful, in case if in the future somebody does refactoring. Also it will help to understand the purpose, if another Code Wrangler will do maintenance of this function